### PR TITLE
[MH3] Implement Wrath of the Skies

### PR DESCRIPTION
**Pull request by DominionSpy**
#11852

### CODE REVIEW

File: Mage.Sets/src/mage/cards/w/WrathOfTheSkies.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+        int xValue = source.getManaCostsToPay().getX();
+        if (xValue < 1) {
```
Comment (Susucre) [inline Mage.Sets/src/mage/cards/w/WrathOfTheSkies.java:66]: Seems wrong for X=0 to return that early.

File: Mage.Sets/src/mage/cards/w/WrathOfTheSkies.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            return true;
+        }
+        new GetEnergyCountersControllerEffect(xValue).apply(game, source);
+
+        int numberToPay = controller.getAmount(0, controller.getCounters().getCount(CounterType.ENERGY),
+                "Pay any amount of {E}", game);
+        if (numberToPay < 1) {
+            return true;
```
Comment (Susucre) [inline Mage.Sets/src/mage/cards/w/WrathOfTheSkies.java:74]: Same. You can pay 0 {E} and still destroy things.

File: Mage.Sets/src/mage/cards/w/WrathOfTheSkies.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        Cost cost = new PayEnergyCost(numberToPay);
+        if (cost.pay(source, game, source, source.getControllerId(), true)) {
+            game.getBattlefield()
+                    .getActivePermanents(controller.getId(), game)
+                    .stream()
+                    .filter(permanent -> permanent.isArtifact(game)
+                            || permanent.isCreature(game)
+                            || permanent.isEnchantment(game))
+                    .filter(permanent -> permanent.getManaValue() <= numberToPay)
+                    .forEach(permanent -> permanent.destroy(source, game, false));
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/w/WrathOfTheSkies.java:81]: just confirming this works fine? I recall potential issues with removing objects from streams, don't know if it is actually a problem here/anymore.

Comment (JayDi85) [reply Mage.Sets/src/mage/cards/w/WrathOfTheSkies.java:81]: Yes, it’s still a problem for streams and any cycle code usages - do not call potentially game state modification code inside original game objects list due ConcurrentModification error. More details in #6207

It’s ok in current use case - getActivePermanents returns new list, so it’s safe to any battlefield changes. But some other game methods can return original list like getStack, getAllPermanents, etc. So it must be used with careful.

**Review by JayDi85 — APPROVED** [review]

**Pull request merged**